src/app/meds-n-labels/activity-1/matchmeds.component.ts

```diff
@@ -0,0 +1,297 @@
+      this.endPos = this.width - (this.radius) - this.lineWidth;
+      this.redrawCanvas();
+    }
+
+    /**
+     * Event gets called when mouse is Released, changes clicked variable to false to prevent drawing.
+     */
+    mouseLeave() {
+        this.clicked = false;
+    }
+
+    /**
+     * Event gets called when mouse is clicked, Starts the mouse drawing.
+     * @param  {number} $event Mouse Coordinates
+     */
+    mouseClick($event) {
+        this.setLineStyle('#9932cc', 10);
+        this.cx.beginPath();
+        this.tempStart.x = $event.clientX - this.canvasLeft;
+        this.tempStart.y = $event.clientY - this.canvasTop;
+        this.cx.moveTo(this.tempStart.x, this.tempStart.y);
+        this.clicked = true;
+    }
+
+    /**
+     * Event gets called when mouse is moved, traces mouse path if mouse is clicked,
+     * @param  {number} $event Mouse Coordinates
+     */
+    mousePos($event) {
+        if (this.clicked) {
+            const x1 = $event.clientX - this.canvasLeft;
+            const y1 = $event.clientY - this.canvasTop;
+            this.cx.lineTo(x1, y1);
+            this.cx.stroke();
+        }
+    }
+
+    /**
+     * Event gets called when mouse is Released, Completes the drawing and checks if valid
+     * @param  {number} $event Mouse Coordinates
+     */
+    mouseRelease($event) {
+        this.tempEnd.x = $event.clientX - this.canvasLeft;
+        this.tempEnd.y = $event.clientY - this.canvasTop;
+        this.cx.lineTo(this.tempEnd.x, this.tempEnd.y);
+        this.cx.stroke();
+        this.clicked = false;
+        this.checkStartAndEnd();
+    }
+
+    /**
+     * Redraws the Canvas
+     */
+    redrawCanvas() {
+      this.cx.clearRect(0, 0, this.width, this.height);
+
+      for (let i = 0; i < this.numElements; i++) {
+          this.setLineStyle('#466bc1', this.lineWidth);
+          this.cx.beginPath();
+          this.cx.arc( this.startPos + this.radius, this.gap * i + this.radius + this.lineWidth, this.radius, 0, 2 * Math.PI );
+          this.cx.stroke();
+          this.cx.beginPath();
+          this.cx.arc( this.endPos , this.gap * i + this.radius + this.lineWidth, this.radius, 0, 2 * Math.PI );
+          this.cx.stroke();
+      }
+
+      for (let i = 0; i < this.count; i ++) {
+          this.cx.beginPath();
+          this.setLineStyle('#9932cc', 10);
+          this.cx.moveTo(this.start[i].x, this.start[i].y);
+          this.cx.lineTo(this.end[i].x, this.end[i].y);
+          this.cx.stroke();
+      }
+    }
+
+    /**
+     * Checks If drawing completes on valid start and end points, checks for right matching
+     */
+
+    checkStartAndEnd() {
+      if ( this.count < this.numElements) {
+        let startValid = false;
+        let endValid = false;
+        let left = 0;
+
+        for (let i = 0; i < this.numElements; i++ ) {
+            if ( this.tempStart.x > this.startPos && this.tempStart.x < this.startPos + 2 * (this.radius)
```

Comment:
create variables for `this.radius * 2`, `this.gap * i` & other possible duplicated arithmetics